content/cli/workspaces.md

```diff
@@ -0,0 +1,207 @@
+
+A mono repo uses the concept of a workspace to manage its member entities. Workspaces are composed of **projects**.  A project may be either:
+- an **application**: a full Nest application including a `main.ts` file to bootstrap the application. Aside from compile and build considerations, an application-type project within a workspace is functionally identical to an application within a *standard mode* structure.
+- a **library**: a library is a way of packaging a general purpose set of features (modules, providers, controllers, etc.) that can be used within other projects.  A library cannot run on its own, and has no `main.ts` file.  Read more about libraries [here](/cli/libraries).
+
+All workspaces have a **default project** (which should be an application-type project).  In the `nest-cli.json` file above, the default project is `my-project`.  This is defined by the top-level `"root"` property, which points at the root of the default project.  Usually, this is the **standard mode** application you started with, and later converted to a monorepo using `nest generate app`.  When you follow these steps, this property is populated automatically for you with information about the standard mode project you started with.
+
+Default projects are used by `nest` commands like `nest build` and `nest start` when a project name is not supplied.
+
+For example, in the above structure, running
+```bash
+$ nest start
+```
+
+will start up the `my-project` app.  To start `my-app`, we'd use:
+
+```bash
+$ nest start my-app
+```
+
+#### Workspace global properties
+
+Workspaces have additional workspace-wide properties that help hide the complexity of working in a multi-project environment.  These are reflected in the following top-level properties of the `nest-cli.json` file:
+
+- `collection`: points at the collection of schematics used to generate components; you generally should not change this value
+- `sourceRoot`: points at the root of the source code for the *default project*
+- `monorepo`: for a monorepo mode structure, this value is always `true`
+- `root`: points at the project root of the *default project*
+- `compilerOptions`: a map with keys specifying compiler options and values specifying the option setting
+
+#### Compiler options
+
+Notice that by default, `compilerOptions` is set as follows:
+```javascript
+"compilerOptions": {
+  "webpack": true,
+  "tsConfigPath": "apps/my-project/tsconfig.app.json"
+}
+```
+
+The `webpack` property, when true (the default), uses [webpack](https://webpack.js.org/) to compile and bundle the code.  This is one key difference between monorepo mode and standard mode.  Standard mode uses `tsc` for compilation by default.  The reason for this difference is that webpack can have significant advantages in build times and in producing a single file bundling all project components together.  If you wish to generate individual files, set `webpack` to `false`, which will cause the build process to use `tsc`.
```

Comment:
We should add that it's possible to define an extra, custom webpack configuration in order to extend defaults (`compilerOptions.webpackConfigPath`)

Example:

```
"compilerOptions": {
  "webpack": true,
  "tsConfigPath": "apps/my-project/tsconfig.app.json",
  "webpackConfigPath": "apps/my-project/webpack.config.js"
}
```

By default, Nest CLI will look for `webpack.config.js` in the root directory.

Example `webpack.config.js` (equal to a regular webpack configuration file):

```javascript
module.exports = {
  externals: []
}
```

(above setting means that webpack will try to bundle `node_modules`, while, by default, it's excluded)

Also, the `webpack.config.js` can expose a `function` that takes default options and returns a modified object.

```javascript
export default function(options) {
   return {
      ...options,
      externals: [],
   }
}
```

Reply:
Great, this is in the updated draft.